### ISSUE TITLE
[QUESTION] Multiple JuiceFS setups in one cluster pitfails?

### ISSUE THREAD

**Issue by TheIronFlo** [juicedata/juicefs-csi-driver#534]
## Question

I build two JuiceFS installations with keydb backend in a (one = 1) kubernetes cluster (1.23.12) in different namespaces. Both are working, but I see mount-pods coming up in the second namespace when I mount volumes from the first storageclass (=first namespace). To make it special: I have mount-pods in both namespaces and can't identify where they belong to.

I changed the names of all objects in the k8s.yaml (e.g. juicefs-csi-controller-sa, juicefs-csi-node-sa, etc.) including CSIDriver so I provide two separate installations of them all in two separate namespaces.

In a mount-pod of namespace 2 I can find env-variable metaurl pointing to the keydb setup in namespace 1, but I provide the correct metaurl to JuiceFS.

So is it possible that there are some pitfails here and we cannot use more than 1 JuiceFS setups in one kubernetes cluster? I can't identify where the mount-pods belong to so I can't provide more details here.

**Comment by TheIronFlo** [comment]
To add a little spice: If I completely remove the 2nd installation including all pods and namespace some mounts crash but with deleting and recreating all deployments they'll run again.

**Comment by zwwhdls** [comment]
Hi @TheIronFlo , 

First of all, you don't need to setup multiple JuiceFS CSI Driver in one cluster. If you want to use JuiceFS volume in different namespaces, you can use PVCs in different namespaces with the same or different storageClasses.

About how to find mount pod where they belong to, please refer to [this doc](https://juicefs.com/docs/csi/troubleshooting/#check-mount-pod)

**Comment by TheIronFlo** [comment]
> Hi @TheIronFlo ,
> 
> First of all, you don't need to setup multiple JuiceFS CSI Driver in one cluster. If you want to use JuiceFS volume in different namespaces, you can use PVCs in different namespaces with the same or different storageClasses.
> 
> About how to find mount pod where they belong to, please refer to [this doc](https://juicefs.com/docs/csi/troubleshooting/#check-mount-pod)

That's right, but if I want to test new versions (or settings changed) of JuiceFS or KeyDB - then I need multiple installations. For the moment I switched back to one (1) setup with multiple storage classes and it works, but in the future I want to test things before use it in production.

**Comment by zwwhdls** [comment]
> > Hi @TheIronFlo ,
> > First of all, you don't need to setup multiple JuiceFS CSI Driver in one cluster. If you want to use JuiceFS volume in different namespaces, you can use PVCs in different namespaces with the same or different storageClasses.
> > About how to find mount pod where they belong to, please refer to [this doc](https://juicefs.com/docs/csi/troubleshooting/#check-mount-pod)
> 
> That's right, but if I want to test new versions (or settings changed) of JuiceFS or KeyDB - then I need multiple installations. For the moment I switched back to one (1) setup with multiple storage classes and it works, but in the future I want to test things before use it in production.

Got it. Did you change all places containing "csi.juicefs.com" in k8s.yaml like https://github.com/juicedata/juicefs-csi-driver/blob/master/deploy/k8s.yaml#L494 also?

**Comment by zxh326** [comment]
need more info
